analyzer/src/main/kotlin/managers/DotNet.kt

```diff
@@ -70,33 +70,61 @@ class DotNet(
  * https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files.
  */
 class DotNetPackageFileReader : XmlPackageFileReader {
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private data class PropertyGroup(
+        @JsonProperty(value = "TargetFramework")
+        val targetFramework: String?,
+        @JsonProperty(value = "TargetFrameworks")
+        val targetFrameworks: String?
+    )
+
     @JsonIgnoreProperties(ignoreUnknown = true)
     private data class ItemGroup(
+        @JacksonXmlProperty(isAttribute = true, localName = "Condition")
+        val condition: String?,
         @JsonProperty(value = "PackageReference")
         @JacksonXmlElementWrapper(useWrapping = false)
-        val packageReference: List<PackageReference>?
+        val packageReferences: List<PackageReference>?
     )
 
     @JsonIgnoreProperties(ignoreUnknown = true)
     private data class PackageReference(
+        @JacksonXmlProperty(isAttribute = true, localName = "Condition")
+        val condition: String?,
         @JacksonXmlProperty(isAttribute = true, localName = "Include")
         val include: String,
         @JacksonXmlProperty(isAttribute = true, localName = "Version")
-        val version: String
+        val version: String,
+        @JsonProperty(value = "PrivateAssets")
+        val privateAssets: String?
     )
 
     override fun getDependencies(definitionFile: File): Set<NuGetDependency> {
+        // TODO: Find a way to parse both collections at once to not have to parse the file twice. The problem with
+        //       adding propertyGroups and itemGroups to a parent object is that the elements can be mixed and in this
+        //       case Jackson only returns the last match for each list.
+        val propertyGroups = NuGetSupport.XML_MAPPER.readValue<List<PropertyGroup>>(definitionFile)
         val itemGroups = NuGetSupport.XML_MAPPER.readValue<List<ItemGroup>>(definitionFile)
 
-        return itemGroups.flatMapTo(mutableSetOf()) { itemGroup ->
-            itemGroup.packageReference?.map { packageReference ->
-                NuGetDependency(
-                    name = packageReference.include,
-                    version = packageReference.version,
-                    targetFramework = "",
-                    developmentDependency = false
-                )
-            }.orEmpty()
+        val targetFrameworks = propertyGroups.find { it.targetFramework != null }?.targetFramework?.let { listOf(it) }
```

Comment:
@mnonnenmacher is it really correct here that only the first non-null `targetFramework` from the list of `propertyGroups` is taken into account, and any `targetFramework` from remaining `propertyGroups` is ignored?

Reply:
Yes, it can only be defined once. If multiple target frameworks they have to be separated by ";".